mne/io/snirf/_snirf.py

```diff
@@ -534,14 +539,25 @@ def _extract_sampling_rate(dat):
     else:
         # specified as time points
         fs_diff = np.around(np.diff(time_data), decimals=4)
+
         if len(np.unique(fs_diff)) == 1:
             # Uniformly sampled data
             sampling_rate = 1.0 / np.unique(fs_diff).item()
         else:
-            warn(
-                "MNE does not currently support reading "
-                "SNIRF files with non-uniform sampled data."
+            mean_sample_difference = np.mean(np.diff(time_data))
+            max_sample_jitter = np.max(np.diff(np.diff(time_data)))
+            sample_jitter_percentage = 100 * (
+                max_sample_jitter / mean_sample_difference
             )
+            if sample_jitter_percentage < MAXIMUM_ALLOWED_SAMPLING_JITTER_PERCENTAGE:
+                # Likely uniformly sampled data with some precision issues.
+                # This is a workaround to provide support for Artinis data.
+                sampling_rate = 1.0 / mean_sample_difference
+            else:
```

Comment:
Otherwise `sampling_rate` will end up as `0` which will be very broken. Better to always set it to as reasonable value as possible I think.
```suggestion
            # Hopefully uniformly sampled data with some precision issues.
            # This is a workaround to provide support for Artinis data.
            sampling_rate = 1.0 / mean_sample_difference
            if sample_jitter_percentage > MAXIMUM_ALLOWED_SAMPLING_JITTER_PERCENTAGE:
```